src/shell/history.rs

```diff
@@ -58,4 +102,89 @@ impl<'a> ShellHistory for Shell<'a> {
             context.history.set_file_name(None);
         }
     }
+
+    fn save_command_in_history(&mut self, command: &str) {
+        if self.should_save_command(command) {
+            // Mark the command in the context history
+            self.set_context_history_from_vars();
+            if let Err(err) = self.context.as_mut().unwrap().history.push(command.into()) {
+                let stderr = io::stderr();
+                let mut stderr = stderr.lock();
+                let _ = writeln!(stderr, "ion: {}", err);
+            }
+        }
+    }
+
+    fn update_ignore_patterns(&mut self, patterns: &Array) {
+        let mut flags = IgnoreFlags::empty();
+        let mut regexes = Vec::new();
+        // for convenience and to avoid typos
+        let regex_prefix = "regex:";
+        for pattern in patterns {
+            match pattern.as_ref() {
+                "all" => flags |= IGNORE_ALL,
+                "no_such_command" => flags |= IGNORE_NO_SUCH_COMMAND,
+                "whitespace" => flags |= IGNORE_WHITESPACE,
+                // The length check is there to just ignore empty regex definitions
+                _ if pattern.starts_with(regex_prefix) && pattern.len() > regex_prefix.len() => {
+                    flags |= IGNORE_BASED_ON_REGEX;
+                    let regex_string = pattern[regex_prefix.len()..].to_owned();
```

Comment:
Maybe only use `to_owned()` when pushing to the `regexes` vector?

Reply:
Actually, we don't need `to_owned()` at all here. It's a surviver of some earlier version and I forgot to remove it when refactoring.